mautic/mautic
# Relative date values are used in segment filters with datetime type

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:


Relative dates are not saved correctly in the segment filter. When we try to create and save a segment with the relative date filter `2 days ago` or `3 months ago`, etc. instead of the relative date will be the regular datetime. This PR solves this problem.

The relative date `first day of` and `last day of` will also display correctly on the segment page. 

For contacts to be filtered, the [PR #10822](https://github.com/mautic/mautic/pull/10822) is needed

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a segment with a datetime filter by the relative date, e.g. "2 days ago". The filter `Date Last Active` can be used
3. Click button "save&close"
4. Go to edit the segment again
5. There is relative date. There was regular date before





<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11375"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
